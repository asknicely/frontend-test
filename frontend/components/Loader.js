import React from 'react';

function Loader() {
  return (
    <div className="loader">
      <div className="loader-dash loader-dash--one"></div>
      <div className="loader-dash loader-dash--two"></div>
      <div className="loader-dash loader-dash--three"></div>
      <div className="loader-dash loader-dash--four"></div>
    </div>
  );
}

export default Loader;
