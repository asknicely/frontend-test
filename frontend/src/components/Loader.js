import React from 'react'

const Loader = () => {
    return (
        <div>
            <div className="preloader">
                <div className="image">
                    <img src="images/loading.gif" alt="Loading ..."></img>
                </div>
                <div className="text">
                    Loading data from database ...
                </div>
            </div>
        </div>
    )
}

export default Loader