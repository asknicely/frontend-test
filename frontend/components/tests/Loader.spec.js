import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Loader from '../Loader';

describe('Loader', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Loader />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Loader correctly', () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  });
});
