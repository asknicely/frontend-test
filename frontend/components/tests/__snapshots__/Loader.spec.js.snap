// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render Loader correctly 1`] = `ShallowWrapper {}`;

exports[`Loader snapshot renders 1`] = `
<div
  className="loader"
>
  <div
    className="loader-dash loader-dash--one"
  />
  <div
    className="loader-dash loader-dash--two"
  />
  <div
    className="loader-dash loader-dash--three"
  />
  <div
    className="loader-dash loader-dash--four"
  />
</div>
`;
